# Problem 1

1) Correct -> as per ANOVA testing

2) Incorrect -> alternate hypothesis is too certain and the null hypothesis is not too specific. 

3) Correct -> population unknown we can perform t-test or z-test (depends on sample size anyway)

4) incorrect -> cannot use p value for hypothesis

5) correct -> sample variance check . Can be done using t-test or z-test

# Problem 2

In [ ]:
mean = 52
sd = 4.5

sample_mean = 52.8
sample_size = 100
sig = 0.05

z_score = (sample_mean - mean) / (4.5/10)
print(z_score)

1.7777777777777715


z score of 0.05 = 0.51994
Z Test >  Z critical
accept null hypothesis
claim of avg 52 is correct

# Problem 3

H0 -> Mean < 34
H1 -> Mean > 34

pop_mean = 34
pop_sd = 8

n =50
sample_mean = 32.5
sig = 0.01

z score = (32.5 - 34) / (8/sqrt(50)) => -1.3258252147247767 => 0.09342

Z(critical) = 0.50399

Z critical > Z test

Hence we reject null hypothesis. The claim is incorrect

# Problem 4

In [ ]:
import numpy as np
sample_mean = np.mean([1008, 812, 1117, 1323, 1308, 1415, 831, 1021, 1287, 851, 930, 730, 699,
872, 913, 944, 954, 987, 1695, 995, 1003, 994])

sample_sd = np.std([1008, 812, 1117, 1323, 1308, 1415, 831, 1021, 1287, 851, 930, 730, 699,
872, 913, 944, 954, 987, 1695, 995, 1003, 994])

print(sample_mean)
print(sample_sd)

1031.3181818181818
234.84799462314425


H0 -> Mean = 1135

H1 -> Mean != 1135

n = 22 

t_score = (1031.318 - 1135) / (234.85/sqrt(22)) = -2.070733177815682

sig = 0.5

df = 21

its a two tail test 

t(test) = 0.686

t(test) > t(critical) 

Hence we accept null hypothesis


# Problem 5

H0 -> mean 48432

H1 -> mean != 48432


mean = 48432

sample mean = 48574

sample_sd = 2000

n = 400


In [ ]:
def z_score(pmean, smean, ssd, n):
  return (pmean - smean)/(ssd/math.sqrt(n))

In [ ]:
z_score(48432,48574,2000,400)

-1.42

Usually the significance we can have is 0.05 (standard) 

Z(test) = 0.07780

Z_test > Z_critical

Hence we accept null hypothesis



# Problem 6

mean = 32.28

n = 19

sample_mean = 31.67

sample_sd = 1.29

H0-> mean = 32.28

H1-> mean != 32.28


In [ ]:
z_score(32.28,31.67,1.29,19)

2.06118477175179

t score = 2.06118477175179

its a 2 tailed test

df = 18

Hence t(critical) = 2.101


t_test < t critical

Reject Null hypothesis


# Problem 7

# Problem 8

pmean = 10

smean = 12

ssd = 1.5

n = 16

In [ ]:
z_score(10 ,12, 1.5, 16) # t score

-5.333333333333333

# Problem 9

In [ ]:
sample_size = 16
df = 15
t_score = 2.947 

# Problem 10

In [ ]:
n = 25
pop_sd = 4
pop_mean = 60
95% of all sample means t score

df = 24
t score = 2.064

t(0.05) = 1.711 t(0.10) = 1.318



In [ ]:
1.711 - 1.318

0.393

# Problem11

Hypothesized difference between population mean d = 0

n1 = 1200, x1 = 452 , s1 = 212

n2 = 800, x2 = 523, s2 = 185

H0-> x1-x2 = 0

H1-> x1-x2 != 0

SE = sqrt[(s1^2/n1) + (s2^2/n2)]

DF = (s1^2/n1 + s2^2/n2)^2 / { [ (s1^2 / n1)^2 / (n1 - 1) ] + [ (s2^2 / n2)^2 / (n2 - 1) ] }

t = [ (x1 - x2) - d ] / SE

In [ ]:
n1 = 1200
x1 = 452
s1 = 212

n2 = 800
x2 = 523
s2 = 185

In [ ]:
import math
import numpy as np

In [ ]:
def bracket_inside(s1,n1,s2,n2):
  first = (s1*s1)/n1
  second = (s2*s2)/n2
  return first,second,first+second

def stdErr(s1,n1,s2,n2):
  _,_,final = bracket_inside(s1,n1,s2,n2)
  return math.sqrt(final)

def dof(s1,n1,s2,n2):
  _,_,numerator = bracket_inside(s1,n1,s2,n2)
  numerator = np.power(numerator,2)
  denom1,denom2,_ =bracket_inside(s1,n1,s2,n2)
  denom1 = denom1/ (n1-1)
  denom2 = denom2/ (n2-1)
  denom = denom1+denom2
  return numerator/denom



In [ ]:
SE = stdErr(s1,n1,s2,n2)
df = dof(s1,n1,s2,n2)

In [ ]:
t = (x1-x2) / SE

In [ ]:
t

-7.926428526759299

In [ ]:
df

75932.29803125162

In [ ]:
from scipy import stats
#Studnt, n=999, p<0.05, 2-tail
#equivalent to Excel TINV(0.05,999)
print(stats.t.ppf(0.05, df))

-1.644873694668252


t test < t critical 
Hence we reject null hypothesis

# Problem 12

In [ ]:

n1 = 100
x1 = 308
s1 = 84

n2 = 100
x2 = 254
s2 = 67

In [ ]:
SE = stdErr(s1,n1,s2,n2)
df = dof(s1,n1,s2,n2)

In [ ]:
t = (x1-x2) / SE

In [ ]:
t

5.025702668336442

In [ ]:
df

11429.550000000001

In [ ]:
from scipy import stats
#Studnt, n=999, p<0.05, 2-tail
#equivalent to Excel TINV(0.05,999)
print(stats.t.ppf(0.05, df))

-1.6449869562087245


t test > t critical

Hence we accept null hypothesis that means of the two batteries are same

# Problem 13

# Problem 14

In [ ]:
n1 = 15
x1 = 6598 
s1 = 844
n2 = 12
x2 = 6870
s2 = 669

In [ ]:
SE = stdErr(s1,n1,s2,n2)
df = dof(s1,n1,s2,n2)

In [ ]:
t = (x1-x2) / SE

In [ ]:
t

-0.9341299576730813

In [ ]:
df

1059850.1319325287

In [ ]:
from scipy import stats
#Studnt, n=999, p<0.05, 2-tail
#equivalent to Excel TINV(0.05,999)
print(stats.t.ppf(0.05, df))

-1.6448550646740732


t test < t critical 
Hence we reject null hypothesis both means are not same

# Problem 15

In [ ]:
n1 = 1000
x1 = 53
p1 = 0.53

n2 = 100
x2 = 43
p2= 0.53

In [ ]:
def p_calculate(n1,p1,n2,p2):
  numerator = n1*p1 + n2*p2
  denom = n1+n2
  return numerator/denom

def calc_z(p1,p2,n1,n2):
  p = p_calculate(n1,p1,n2,p2)
  numerator = p1-p2
  denom1 = (p * (1-p))/n1 
  denom2 = (p * (1-p))/n2
  denom = denom1 + denom2
  denom = math.sqrt(denom)
  return numerator/denom

In [ ]:
z_stats = calc_z(p1,p2,n1,n2)
print(z_stats)

0.0


z(critical 0.05) = 0.51994

Z test < Z critical

Hence we reject null hypothesis
Mean not equal

Correction p1 and p2 should be x/n . Hence p2 is given wrong I guess

# Problem 16

In [ ]:
n1 = 300
x1 = 120
p1 = 0.40
n2 = 700
x2 = 140
p2= 0.20

In [ ]:
z_stats = calc_z(p1,p2,n1,n2)
print(z_stats)

6.607499047222305


z test > z critical 

Hence we accept null hypothesis

# Problem 17

H0 -> die is unbiased

H1 -> die is not unbiased

for an unbiased die each face should occur 132/6  = 22 times

Expected value is 22 and observed value is given in frequency

In [ ]:
Observed = [16, 20, 25, 14, 29, 28]
Expected = [22, 22, 22, 22, 22, 22]

In [ ]:
obs_minus_exp_div_exp = [np.power((a-b),2)/b for a,b in zip(Observed,Expected)]

In [ ]:
obs_minus_exp_div_exp

[1.6363636363636365,
 0.18181818181818182,
 0.4090909090909091,
 2.909090909090909,
 2.227272727272727,
 1.6363636363636365]

In [ ]:
np.sum(obs_minus_exp_div_exp)

9.0

dof = 5
sig = 0.05
chi value = 11.070

chi test < chi critical 

Hence we reject null hypothesis

Die is not unbiased

# Problem 18

H0 -> gender is indept of voting

H1 -> gender is not indept of voting

We need to create expected table to apply chi square

In [ ]:
observed_men_voted = 2792
observed_mean_not_voted = 1486

observed_female_voted = 3591
observed_female_not_voted = 2131


In [ ]:
Expected_men_voted = int(((2792+1486) * (2792+3591)) / 10000)
Expected_mean_not_voted = int(((2792+1486) * (1486+2131)) / 10000)

Expected_female_voted = int(((3591+2131) * (2792+3591)) / 10000)
Expected_female_not_voted = int(((3591+2131) * (1486+2131)) / 10000)

print(Expected_men_voted,Expected_mean_not_voted,Expected_female_voted,Expected_female_not_voted)

2730 1547 3652 2069


In [ ]:
observed = [2792,1486,3591,2131]
expected = [2730,1547,3652,2069]

In [ ]:
obs_minus_exp_div_exp = [np.power((a-b),2)/b for a,b in zip(observed,expected)]

In [ ]:
obs_minus_exp_div_exp

[1.4080586080586082, 2.40530058177117, 1.018893756845564, 1.8579023682938618]

In [ ]:
np.sum(obs_minus_exp_div_exp)

6.6901553149692035

for dof 1 and 2 with sig 0.05, 3.841 and 5.991

chi test > chi critical

Hence accept null hypothesis

Voters are independant

# Problem 19

In [ ]:
observed = [41,19,24,16]
expected = [25,25,25,25]

In [ ]:
obs_minus_exp_div_exp = [np.power((a-b),2)/b for a,b in zip(observed,expected)]

In [ ]:
obs_minus_exp_div_exp

[10.24, 1.44, 0.04, 3.24]

In [ ]:
np.sum(obs_minus_exp_div_exp)

14.959999999999999

chi test 14.96

chi critical for dof = 3 and p<0.05

chi critical for p< 0.05 are 7.815	9.348	11.345	12.838

In [ ]:
np.mean([7.815,9.348,11.345,12.838])

10.336500000000001

chi test > chi critical

Hence we accept null hypothesis

suggest that voters do not prefer the four candidates equally. 

# Problem 20

In [ ]:
import numpy as np

In [ ]:
observed = np.array([[18,22,20],
                    [2,28,40],
                    [20,10,40]])

In [ ]:
column_total = np.sum(observed,axis=0)
row_total = np.sum(observed,axis=1)

In [ ]:
grand_total = np.sum(observed)

In [ ]:
# row_total * column_total / grand_total

In [ ]:
observed = [18,22,20,2,28,40,20,10,40]
expected = [12,18,30,14,21,35,14,21,35]

In [ ]:
obs_minus_exp_div_exp = [np.power((a-b),2)/b for a,b in zip(observed,expected)]

In [ ]:
obs_minus_exp_div_exp

[3.0,
 0.8888888888888888,
 3.3333333333333335,
 10.285714285714286,
 2.3333333333333335,
 0.7142857142857143,
 2.5714285714285716,
 5.761904761904762,
 0.7142857142857143]

In [ ]:
np.sum(obs_minus_exp_div_exp)

29.603174603174608

chi critical = 9.488
chi test > chi critical
Hence we accept null hypothesis

# Problem 21

Chi-Square = 19.87

df = 1
p < 0.05

5.024	6.635	7.879

chi critical 6.512666666666665  < Chi test

Hence accept null hypothesis

# Problem 22

Chi-Square = 10.71

with 2 df

p < 0.01

chi critical = 9.210	

chi test >  chi critical 

accept null hypothesis

# Problem 23

In [ ]:
observed = [679, 103, 114, 63, 10, 20, 42, 18,25]

In [ ]:
expected = [654, 109, 133, 68, 11, 14, 62, 10, 13]

In [ ]:
obs_minus_exp_div_exp = [np.power((a-b),2)/b for a,b in zip(observed,expected)]

In [ ]:
obs_minus_exp_div_exp

[0.9556574923547401,
 0.3302752293577982,
 2.7142857142857144,
 0.36764705882352944,
 0.09090909090909091,
 2.5714285714285716,
 6.451612903225806,
 6.4,
 11.076923076923077]

In [ ]:
np.sum(obs_minus_exp_div_exp)

30.958739137308328

In [ ]:
dof = (3-1) * (3-1)

In [ ]:
dof

4

assume sig = 0.05

chi test > chi critical 9.488

Hence we accept null hypothesis